fabric/src/main/java/fr/raksrinana/fallingtree/fabric/tree/breaking/ToolDamageHandler.java

```diff
@@ -31,8 +31,18 @@ private int getDamage(long count){
 			case ROUND_DOWN -> Math.floor(rawDamage);
 			case ROUND_UP -> Math.ceil(rawDamage);
 			case ROUNDING -> Math.round(rawDamage);
+			case PROBABILISTIC -> this.getProbabilisticDamage(rawDamage);
 		};
 	}
+
+	private int getProbabilisticDamage(double rawDamage) {
+		int damage = (int)Math.floor(rawDamage);
+		double probability = rawDamage - Math.floor(rawDamage);
```

Comment:
Probably can use `damage` instead of calculating the floor a second time.
```suggestion
		double probability = rawDamage - damage;
```

Reply:
I decided to go against it because of constant type recasting. It might introduce discrepancies and errors on conversion. Although I don't know Java well enough to know whether it's actually the case, so if you believe there can't be any errors, second floor calculation is unnecessary indeed.

Reply:
There can be some errors when doing this, but it's tiny errors that don't really have much impact for us. Like if there's a 0.001 error in the conversion that isn't too much of a problem, we're not dealing with money or sensitive data.

Also floor performs quite a lot under the hood I think it's less costly to have it convert `int -> double` again rather than redoing the floor.

As we know we're dealing with positive values we could also avoid the floor altogether and do more "direct" operations:
```java
int damage = (int) rawDamage;
double probability = rawDamage % 1; // Will have to check this though
```

I let it up to you if wanna do the change or not. The performance doesn't really matter actually as this method isn't called often. Just if you wanna keep the double, put it at least in a variable to avoid performing all the floor calculations twice.